File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/DungeonPartyFinderAvailableDungeonsMessage.scala





















// Generated on 05/03/2013 10:34:46
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait DungeonPartyFinderAvailableDungeonsMessage extends MessageDeserializer {
  val messageId = 6242

  protected trait Message extends BaseMessage with Serializable {
    val messageId = DungeonPartyFinderAvailableDungeonsMessage.messageId

    val dungeonIds: Seq[Short]

    def serialize(buf: Buffer) {
      buf.writeUShort(dungeonIds.length)
      for (entry <- dungeonIds) {
        buf.writeShort(entry)
      }
    }

  }

  protected def create(dungeonIds: Seq[Short]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val dungeonIds = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }

    create(dungeonIds)
  }

}

object DungeonPartyFinderAvailableDungeonsMessage extends DungeonPartyFinderAvailableDungeonsMessage {
  case class MessageImpl(dungeonIds: Seq[Short]) extends Message
  type Target = MessageImpl

  override def create(dungeonIds: Seq[Short]) = MessageImpl(dungeonIds)

  def apply(dungeonIds: Seq[Short]) = create(dungeonIds)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/DungeonPartyFinderAvailableDungeonsMessage.java


// Generated on 05/08/2013 19:37:48
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class DungeonPartyFinderAvailableDungeonsMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6242;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short[] dungeonIds;
    
    public DungeonPartyFinderAvailableDungeonsMessage() { }
    
    public DungeonPartyFinderAvailableDungeonsMessage(short[] dungeonIds) {
        this.dungeonIds = dungeonIds;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(dungeonIds.length);
        for (short entry : dungeonIds) {
            buf.writeShort(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        dungeonIds = new short[limit];
        for (int i = 0; i < limit; i++) {
            dungeonIds[i] = buf.readShort();
        }
    }
    
}
